File: JAVA/66. Plus One.java
class Solution {
    public int[] plusOne(int[] d) {
        d[d.length-1]++;
        for(int i=d.length-1;i>=0;i--)
        {
            if(d[i]==10 && i!=0)
            {
                d[i-1]++;
                d[i]=0;
                continue;
            }
            else
            break;
        }
        if(d[0]==10)
        {
            d[0]=0;
            int[] a=new int[d.length+1];
            for(int i=0;i<d.length;i++)
            {
                a[i+1]=d[0];
            }
            a[0]=1;
            return a;
        }
        return d;
    }
}

File: C++/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& d) {
        d[d.size()-1]++;
        for(int i=d.size()-1;i>=0;i--)
        {
            if(d[i]==10 && i!=0){
                d[i-1]++;
                d[i]=0;
            continue;}
            else
            break;
        }
        if(d[0]==10)
        {
            d[0]=0;
            d.insert(d.begin(),1);
        }
        return d;
    }
};